app/helpers/request_dialog_options_helper.rb

```diff
@@ -0,0 +1,66 @@
+module RequestDialogOptionsHelper
+  def request_dialog_options(workflow)
+    data = workflow.dialog.dialog_tabs.map do |tab, _tab_index|
+      {
+        :label   => _(tab.label),
+        :content => request_dialog_options_groups(tab.dialog_groups, workflow)
+      }
+    end
+    react('RequestDialogOptions', {:data => data})
+  end
+
+  private
+
+  def request_dialog_options_groups(dialog_groups, workflow)
+    dialog_groups.map do |group|
+      {
+        :title => _(group.label),
+        :rows  => request_dialog_options_fields(group.dialog_fields, workflow),
+        :mode  => "miq_summary request_dialog_options"
+      }
+    end
+  end
+
+  def request_dialog_options_fields(dialog_fields, workflow)
+    dialog_fields.map do |field|
+      row_data(_(field.label), request_dialog_options_field_content(field, workflow))
+    end
+  end
+
+  def request_dialog_options_field_content(field, workflow)
+    case field.type
+    when 'DialogFieldTextBox', 'DialogFieldTextAreaBox'
+      if field.protected?
+        "********"
+      else
+        field.value
+      end
+
+    when 'DialogFieldCheckBox'
+      {:input => "checkbox", :name => field.name, :checked => field.checked?, :disabled => true, :label => ''}
+
+    when 'DialogFieldDateControl'
+      field.value
+
+    when 'DialogFieldDateTimeControl'
+      date_val, time_val = field.value.split
+      hour_val, minute_val = time_val.split(":")
+      "#{date_val} at #{hour_val.rjust(2, '0')}:#{minute_val.rjust(2, '0')} #{session[:user_tz]}"
+
+    when "DialogFieldRadioButton"
+      field.values.detect { |k, _v| k == workflow.value(field.name) }.try(:last) || workflow.value(field.name)
+
+    when "DialogFieldDropDownList"
+      field.value.blank? ? _("<None>") : field.value.to_s.gsub("\x1F", ", ")
+
+    when 'DialogFieldTagControl'
+      value = workflow.value(field.name) || '' # it returns in format Clasification::id
+      classifications = value.split(',').map { |c| Classification.find_by(:id => c.split('::').second).description }
+      if classifications.empty?
+        ''
+      else
+        classifications.join(', ')
+      end
+    end
```

Comment:
Sorry, I didn't see this sooner.  This looks like a case we can do [tell don't ask](https://thoughtbot.com/blog/tell-dont-ask). It looks like we're asking objects their type and using conditional logic to figure out the "content".  Instead, we can tell the object to provide it's content and that logic can live in the class instead of here in a long conditional.


Consider instead, updating each class with the desired method name, such as: 

```ruby
class DialogFieldDateControl < DialogField
  def content(_workflow)
    value
  end
...

class DialogFieldTextBox < DialogField
  def content(_workflow)
    protected? ? "********" : super
  end
...

# no need to define it for DialogFieldTextAreaBox since it's a subclass of DialogFieldTextBox

class DialogFieldCheckBox
  def content(_workflow)
    {:input => "checkbox", :name => name, :checked => checked?, :disabled => true, :label => ''}
  end
...

class DialogFieldRadioButton
  def content(workflow)
    values.detect { |k, _v| k == workflow.value(name)) }.try(:last) || workflow.value(name))
...
```

With this code in place and living in the classes, we can change this method to:

```ruby
  def request_dialog_options_field_content(field, workflow)
    field.content(workflow)
  end
```

Reply:
Hey @jrafanie , we just had a discussion on this. And we decided to give this a new UI pasted in the description of https://github.com/ManageIQ/manageiq-ui-classic/pull/8911

Reply:
Sorry, I'm not sure what I'm looking at in #8911.  I'm fine if you do the abstraction in javascript.  My point was to hide information about each type of ui element in their code and not in the caller code.

Reply:
@jeffibm I didn't mean to derail this PR.  I'm fine with moving forward and only meant to ask if the class specific logic in the case statement of `request_dialog_options_field_content` was something we can try to avoid by having the classes knowing this logic, instead of the presentation layer knowing it.  We can move forward but it seemed like something worthy of checking into.